lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -45,3 +45,120 @@ function wp_theme_clean_theme_json_cached_data() {
 		WP_Theme_JSON_Resolver_Gutenberg::clean_cached_data();
 	}
 }
+
+/**
+ * Returns the stylesheet resulting of merging core, theme, and user data.
+ *
+ * @param array $types Types of styles to load. Optional.
+ *                     It accepts 'variables', 'styles', 'presets' as values.
+ *                     If empty, it'll load all for themes with theme.json support
+ *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
+ *
+ * @return string Stylesheet.
+ */
+function gutenberg_get_global_stylesheet( $types = array() ) {
+	/**
+	 * Filters whether the cached global stylesheet can be used.
+	 *
+	 * @since 14.6.0
+	 *
+	 * @param boolean $can_use_cached Whether the cached global stylesheet can be used.
+	 */
+	$can_use_cached = apply_filters(
+		'global_stylesheet_can_use_cache',
+		( empty( $types ) ) &&
```

Comment:
Idea for a potential follow-up: this code currently only caches the whole stylesheet, meaning the one that is enqueued in the front-end ([see](https://github.com/WordPress/gutenberg/blob/f19753448ff73e94ea139507f520c0318b1be82d/lib/compat/wordpress-6.1/script-loader.php#L63)). I wonder whether it makes sense to cache the others for the editor as well ([see](https://github.com/WordPress/gutenberg/blob/f19753448ff73e94ea139507f520c0318b1be82d/lib/compat/wordpress-6.1/block-editor-settings.php#L77)).